### PR TITLE
chore(tests): added some more metadata an private settings tests

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

**Comment by jdalsem** [comment]
Going to fix #11427 with this PR... first added some tests to see if i can reproduce the problem

File: engine/tests/phpunit/integration/Elgg/Integration/ElggCoreGetEntitiesFromPrivateSettingsTest.php [hunk 1]
```diff
@@ -92,5 +92,73 @@ public function testElggApiGettersEntitiesFromPrivateSettings() {
...
+	 * @dataProvider booleanPairsProvider
+	 */
+	public function testElggGetEntitiesFromBooleanPrivateSettings($value, $query, $type) {
+		$object = $this->createOne('object');
+		$object->setPrivateSetting('private_setting', $value);
+		
+		// full pair
+		$options = [
+			'type' => 'object',
+			'subtype' => $this->object->subtype,
```
Comment (jeabakker) [inline engine/tests/phpunit/integration/Elgg/Integration/ElggCoreGetEntitiesFromPrivateSettingsTest.php]: does this match `$object->subtype` or did you mean to do `$object->subtype`

File: engine/tests/phpunit/integration/Elgg/Integration/ElggCoreGetEntitiesFromPrivateSettingsTest.php [hunk 2]
```diff
@@ -92,5 +92,73 @@ public function testElggApiGettersEntitiesFromPrivateSettings() {
...
+		];
+		
+		$result = elgg_get_entities($options);
+		
+		$this->assertEquals(1, $result);
+
+		// short pair
+		$options = [
+			'type' => 'object',
+			'subtype' => $this->object->subtype,
```
Comment (jeabakker) [inline engine/tests/phpunit/integration/Elgg/Integration/ElggCoreGetEntitiesFromPrivateSettingsTest.php]: same as above

File: engine/tests/phpunit/integration/Elgg/Integration/ElggCoreGetEntitiesFromPrivateSettingsTest.php [hunk 3]
```diff
@@ -92,5 +92,73 @@ public function testElggApiGettersEntitiesFromPrivateSettings() {
...
+		];
+		
+		$result = elgg_get_entities($options);
+		
+		$this->assertEquals(1, $result);
+
+		// names and values
+		$options = [
+			'type' => 'object',
+			'subtype' => $this->object->subtype,
```
Comment (jeabakker) [inline engine/tests/phpunit/integration/Elgg/Integration/ElggCoreGetEntitiesFromPrivateSettingsTest.php]: same

**Review by jeabakker — DISMISSED** [review]

**Comment by jdalsem** [comment]
setting a private setting value is always sanitized to a string....

**Comment by jdalsem** [comment]
holy *** .... private settings is terrible.... saving `false` results in `''` value in db. Querying using ege `false` is changed to `0`... this PR makes sure the behaviour is intact with Elgg 2.3, but this is terrible. It also makes merging the private settings into metadata a bigger challenge

**Comment by hypeJunction** [comment]
Hence it's good to refactor code to find all these bugs and behaviours hidden behind thousands of lines of untested code...

**Comment by jdalsem** [comment]
currently you can only use `false` for `elgg_get_entities` from private settings if the entity has save with a `0`... that is why `ElggUpgrade` does the init of `is_completed` to `0` instead of false

**Comment by jdalsem** [comment]
> Hence it's good to refactor code to find all these bugs and behaviours hidden behind thousands of lines of untested code...

next time... add tests of current usage before refactoring... it is very hard to track down when you are not used to the new code yet...

**Comment by jdalsem** [comment]
need to redo this now privatesettings are in line with metadata

**Pull request closed without merging**